ObjectCreation/G3d.FCMacro

```diff
@@ -0,0 +1,475 @@
+        "Offset",
+        "Offset from the sketch plane from which the extrusion along the X axis will start",
+        "Length",
+        "Total length of the lateral extrusion along the X axis (it must be always larger than the lateral offset)",
+        "Offset",
+        "Offset from the sketch plane from which the extrusion along the Z axis will start",
+        "Length",
+        "Total length of the vertical extrusion along the Z axis (it must be always larger than the vertical offset)",
+        "3D-Part",
+        "3D-Part_Views"
+    ],
+    [ ## Stringhe per la traduzione in qualsiasi lingua (stringhe in italiano)
+
+        "Generatore 3D",
+        "Seleziona uno schizzo in 2 o 3 viste:",
+        "Vista frontale - (Fronte/Retro):",
+        "Spunta questa casella se desideri utilizzare la vista frontale ed assegnarle uno schizzo",
+        "Vista laterale - (Destra/Sinistra):",
+        "Spunta questa casella se desideri utilizzare la vista laterale ed assegnarle uno schizzo",
+        "Vista in pianta - (Alto/Basso):",
+        "Spunta questa casella se desideri utilizzare la vista in pianta ed assegnarle uno schizzo",
+        "Tol. 3D (mm.):",
+        "Lunghezza in mm di sovraestrusione. Non dovrebbe influenzare l'oggetto finale. Se non si necessita questa tolleranza impostare il valore relativo a 0 mm",
+        "Crea un nuovo gruppo (cartella) con le viste (sketch)",
+        "Nasconde le viste quando viene generato il solido",
+        "Fusione (intersezione)",
+        "Versione:",
+        "Breve guida all'uso dello script:",
+        "Crea uno sketch per ogni vista necessaria a generare il solido. Sono necessari almeno due sketch.",
+        "Quindi clicca sull'icona dello script e seleziona uno schizzo per ciascuna vista necessaria.",
+        "Infine, premi il pulsante OK per generare il solido.",
+        "Documentazione:",
+        "Offset",
+        "Distanza rispetto al piano dello sketch frontale da cui inizierà l'estrusione lungo l'asse Y (spostamento in Y)",
+        "Lunghezza",
+        "Lunghezza totale dell'estrusione frontale lungo l'asse Y (dovrà essere sempre più grande dell'offset frontale)",
+        "Offset",
+        "Distanza rispetto al piano dello sketch laterale da cui inizierà l'estrusione lungo l'asse X (spostamento in X)",
+        "Lunghezza",
+        "Lunghezza totale dell'estrusione laterale lungo l'asse X (dovrà essere sempre più grande dell'offset laterale)",
+        "Offset",
+        "Distanza rispetto al piano dello sketch in pianta da cui inizierà l'estrusione lungo l'asse Z (spostamento in Z)",
+        "Lunghezza",
+        "Lunghezza totale dell'estrusione verticale lungo l'asse Z (dovrà essere sempre più grande dell'offset in pianta)",
+        "Pezzo3D",
+        "Pezzo3D_Viste"
+    ]
+        ]
+
+
+
+rutaMacro = FreeCAD.ParamGet("User parameter:BaseApp/Preferences/Macro")     
```

Comment:
Please write all the code in English. This is otherwise not maintainable. Please ask for help (I may try to find some time) if this is too much work for your.

Reply:
Hi Galou,

I'm often maintaining the code, and the problem with putting it in English is that then who has a hard time maintaining it is me ;). I could put global variables and generic stuff in English if that's okay with you. The things that help me to understand what I'm doing, I prefer to keep them as they are for now.

For example: rutaMacro --> pathMacro is ok. But "maximoPZ" it would be a real mess for me because the change for "maxTZ" (it makes me confused)

Reply:
The functions (def) could also change them to English... no problem with this

Reply:
Translate global var & def. I hope it's enough

Reply:
It looks to me that your English is good enough to write code completely in English but I understand your point of view. I'll integrate your code as it is.